server.js

```diff
@@ -55,14 +62,20 @@ class Metadata {
                     }).join(' ');
                     doc.scopes = e.scope.split(",");
                 }
-                self.builder.add(doc);
+                if (args.redis) {
+                    this.redis_idx.add(doc);
```

Comment:
With a class approach you can simply set this.idx to an instance of the index class and call the add method.